File: 2020/Python/day5.py
from aoc import *

def solve(inp):
    seats = inp.replace("F", "0").replace("B", "1").replace("L", "0").replace("R", "1")
    seats = seats.splitlines()
    seats = {int(x, 2) for x in seats}
    part1 = max(seats)
    for s in seats:
        if s+2 in seats and s+1 not in seats:
            part2 = s+1
    return part1, part2


run(solve)

File: 2020/Nim/day5.nim
import aoc
import strutils, sequtils

func solve*(input:string): (int, int) =
    let data = input.replace("F", "0").replace("B", "1").replace("L", "0").replace("R", "1")
    var seats: set[int16]
    for s in data.splitLines:
        seats.incl(fromBin[int16](s))
    let part1 = seats.toSeq.max
    for s in seats:
        if s+2 in seats and s+1 notin seats:
            return (int(part1), int(s+1))
                